pennylane_lightning_gpu/src/tests/mpi/Test_MPIManager.cpp

```diff
@@ -0,0 +1,220 @@
+#include <algorithm>
+#include <complex>
+#include <iostream>
+#include <limits>
+#include <type_traits>
+#include <utility>
+#include <vector>
+
+#include <catch2/catch.hpp>
+
+#include "MPIManager.hpp"
+
+#include "../TestHelpers.hpp"
+
+using namespace Pennylane;
+using namespace Pennylane::MPI;
+using namespace CUDA;
+
+TEMPLATE_TEST_CASE("MPIManager::Scatter", "[MPIManager]", float, double) {
+    using PrecisionT = TestType;
+    using cp_t = std::complex<PrecisionT>;
+
+    MPIManager mpi_manager(MPI_COMM_WORLD);
+
+    int rank = mpi_manager.getRank();
+    int size = mpi_manager.getSize();
+
+    SECTION("Apply scatter") {
+        std::vector<cp_t> sendBuf(size);
+        int root = 0;
+        cp_t result(2.0 * rank, 2.0 * rank + 1);
+        if (rank == root) {
+            for (size_t i = 0; i < sendBuf.size(); i++) {
+                cp_t data(2.0 * i, 2.0 * i + 1);
+                sendBuf[i] = data;
+            }
+        }
+
+        auto recvBuf = mpi_manager.scatter<cp_t>(sendBuf, root);
+        CHECK(recvBuf[0].real() == result.real());
+        CHECK(recvBuf[0].imag() == result.imag());
+    }
+
+    SECTION("Apply Scatter") {
+        std::vector<cp_t> sendBuf(size);
+        std::vector<cp_t> recvBuf(1);
+        int root = 0;
+        cp_t result(2.0 * rank, 2.0 * rank + 1);
+        if (rank == root) {
+            for (size_t i = 0; i < sendBuf.size(); i++) {
+                cp_t data(2.0 * i, 2.0 * i + 1);
+                sendBuf[i] = data;
+            }
+        }
+
+        mpi_manager.Scatter<cp_t>(sendBuf, recvBuf, root);
```

Comment:
This will fail if the previous test is not performed first. I think there are a few caveats like in the tests, depends on intended usage. I'll leave comments about how to improve robustness in  `MPIManager.hpp`.

Reply:
I thought `recvBuf` was allocated at line 39 when `auto recvBuf = mpi_manager.scatter<cp_t>(sendBuf, root);`, but I read that wrong.